### PR TITLE
[Enchance] Set a random seed when the user does not set a seed.

### PR DESCRIPTION
**Pull request by wHao-Wu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.
Set a random seed when the user does not set a seed.

## Modification
Please briefly describe what modification is made in this PR.
Add an `init_random_seed` function and call this function when training.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by ZCMax — APPROVED** [review]

**Review by Tai-Wang — APPROVED** [review]

File: mmdet3d/apis/train.py [hunk 1]
```diff
@@ -1,8 +1,44 @@
...
+import numpy as np
+import torch
+from mmcv.runner import get_dist_info
+from torch import distributed as dist
+
 from mmdet.apis import train_detector
 from mmseg.apis import train_segmentor
 
 
+def init_random_seed(seed=None, device='cuda'):
```
Comment (ZwwWayne) [inline mmdet3d/apis/train.py:11]: import from mmdet
